### PR TITLE
Mock: Advanced Space handling + tests

### PR DESCRIPTION
**Pull request by ddd-mtl**
## PR summary
Solved #147 

Implemented handling of Lib3hClientProtocol::LeaveSpace.

Tests for leaving and rejoining a space.
Tests with multiple space.

Refactored some test code easy re-use in other tests.

### CODE REVIEW

File: crates/lib3h/src/engine/real_engine.rs [hunk 1]
```diff
@@ -555,6 +554,28 @@ impl<T: Transport, D: Dht> RealEngine<T, D> {
...
+            to_agent_id: join_msg.agent_id.clone(),
+            result_info: vec![],
+        };
+        // Bail if space not joined by agent
+        let chain_id = (join_msg.space_address.clone(), join_msg.agent_id.clone());
+        if !self.space_gateway_map.contains_key(&chain_id) {
+            res.result_info = "Agent is not part of the space".to_string().into_bytes();
+            return Lib3hServerProtocol::FailureResult(res);
+        }
+        self.space_gateway_map.remove(&chain_id).unwrap();
```
Comment (struktured) [inline crates/lib3h/src/engine/real_engine.rs]: Minor optimization but you could just call `remove` and check if the return value is `None`- if it is then produce the error result. Also, this code is simple enough that using a mutable reference for the result instance isn't a huge deal but it would be really easy to make this code immutable by building the result at the very end and using a conditional let statement to populate the `result_info` .

File: rustfmt.toml [hunk 2]
```diff
@@ -28,7 +28,7 @@ combine_control_expr = true
 struct_field_align_threshold = 0
 match_arm_blocks = true
 force_multiline_blocks = false
-fn_args_density = "Tall"
+fn_args_layout = "Tall"
```
Comment (struktured) [inline rustfmt.toml]: Looks innocuous but not sure why this change is in this PR.

Comment (ddd-mtl) [reply rustfmt.toml]: Minor rust version update leftovers.

File: crates/lib3h/src/engine/real_engine.rs (hunk 1 above)
Comment (ddd-mtl) [reply crates/lib3h/src/engine/real_engine.rs]: @struktured Yeah, its a bit cleaner like you say even if the method is still not immutable.
